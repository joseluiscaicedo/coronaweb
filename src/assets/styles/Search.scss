@import "./Vars.scss";
.search {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 300px;
  border-radius: 0px 0px 40px 40px;
  color:$third-color;
  font-size: 16px;
}
.search__label{
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  flex-wrap: wrap;
  gap: 1rem;
}

.search__title {
  font-size: 20px;
  padding-bottom: 2rem;
  text-align: center;
}

.label__input {
  background-color: rgba(255,255,255,0.1);
  border: 2px solid $third-color;
  border-radius: 25px;
  font-size: 16px;
  height: 50px;
  outline: none;
  padding: 20px;
  width: 50%;
  color:$third-color;
}

.ShowCountries{
  display: grid;
  grid-gap: 1rem;
  place-items: center;
  grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));

}