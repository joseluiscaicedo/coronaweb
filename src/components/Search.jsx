/* eslint-disable react/display-name */
import React, { useState } from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import CarouselItem from './CarouselItem';
import useFetchData from '../hooks/useFetchData';
import Loading from './Loading';

import '../assets/styles/Search.scss';

const APIURL = process.env.APIURL;

const SearchTerm = ({setSearchCountries})=>(event)=> {
  setSearchCountries(event.target.value)
}
const filterTerm = ({ searchCountries }) => (event) =>{
  if(searchCountries==='')return event;
  if (event.Country.toLowerCase().includes(searchCountries.toLowerCase())) {
    return event;
  }
};
const mapCountries =({ CarouselItem }) => ({ ...CovidCountries }) => <CarouselItem key={CovidCountries.ID} {...CovidCountries}/>


const Search = () => {
  const { isAuthenticated } = useAuth0();
  const CovidCountries = useFetchData(APIURL)
  const [searchCountries, setSearchCountries] = useState('');

  return(
    <section aria-label='Container Search' className='ContainerSearch'>
    {CovidCountries.length === 0 ? <Loading /> : (
      <>
        {isAuthenticated ? (
        <>
          <section className='ContainerSearch__search'>
            <h1 aria-label='search title' className='search__title'>Search by Country</h1>
            <label
            aria-labelledby='WriteaCountry'
            className='search__label'
            id='Search'
            htmlFor='WriteaCountry'
            rol='tab'
            > Write a Country
              <input
              id='WriteaCountry'
              name='input'
              type='text'
              className='label__input'
              autoComplete = 'Write a country'
              aria-label='WriteCountry'
              rol='tab'
              onChange={SearchTerm({ setSearchCountries })}
              />
            </label>
          </section>
          <section className='ShowCountries'>
            {CovidCountries.filter(filterTerm({ searchCountries })).map(mapCountries({ CarouselItem }))}
          </section>
        </>):
        (<a aria-label="GoToHome" href='/'>
        You should Login to see this section, Take me to Go To Home
      </a>)
        };
      </>
    )}
    </section>
    );
}

export default Search;
