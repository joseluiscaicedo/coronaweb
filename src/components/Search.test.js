import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import {  render } from '@testing-library/react'
import Search from './Search';

jest.mock('@auth0/auth0-react', () => ({
  useAuth0: ()=>{return {'isAuthenticated':true}},
}))

jest.mock('../hooks/useFetchData', () =>({
  __esModule: true,
  default: ()=>{return [{}]}
}))


describe('validate Search Component', () =>{

  test('Generates Snapshot for Search Component', () => {
    const { container } = render(
      <Search />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Validate by Text', () => {

    const component =render( <Search /> );
    const HeadingElement=component.getByText('Search by Country')
    expect(HeadingElement).toBeInTheDocument();
  });
})