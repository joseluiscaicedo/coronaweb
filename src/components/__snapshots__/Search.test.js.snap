// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Search Component Generates Snapshot for Search Component 1`] = `
<section
  aria-label="Container Search"
  class="ContainerSearch"
>
  <section
    class="ContainerSearch__search"
  >
    <h1
      aria-label="search title"
      class="search__title"
    >
      Search by Country
    </h1>
    <label
      aria-labelledby="WriteaCountry"
      class="search__label"
      for="WriteaCountry"
      id="Search"
      rol="tab"
    >
       Write a Country
      <input
        aria-label="WriteCountry"
        autocomplete="Write a country"
        class="label__input"
        id="WriteaCountry"
        name="input"
        rol="tab"
        type="text"
      />
    </label>
  </section>
  <section
    class="ShowCountries"
  >
    <div
      class="carousel-item"
    >
      <div
        class="carousel-item__details"
        id="details"
      >
        <p
          class="carousel-item__details--title"
        />
        <p
          aria-label="Total Confirmed"
          class="carousel-item__details--subtitle"
        >
          Total Confirmed 
          <br />
          undefined
        </p>
        <p
          aria-label="Total Deaths"
          class="carousel-item__details--subtitle"
        >
          Total Deaths 
          <br />
          undefined
        </p>
        <p
          aria-label="Total Recovered"
          class="carousel-item__details--subtitle"
        >
          Total Recovered 
          <br />
          undefined
        </p>
      </div>
    </div>
  </section>
  ;
</section>
`;
